File: front-todo/src/models/Task.ts

export class Task {
    uuid: string;
    title: string;
    description: string;
    priority: string;
    categoryName: string;
    status: string;
    dueDate: Date;
    reminderDateTime: Date;
    createdAt: string;
    completed:boolean;

    constructor(data: Partial<Task>) {
        this.uuid = data.uuid ?? '';
        this.title = data.title ?? '';
        this.description = data.description ?? '';
        this.priority = data.priority ?? '';
        this.categoryName = data.categoryName ?? '';
        this.status = data.status ?? '';
        this.dueDate = data.dueDate ?? new  Date();
        this.reminderDateTime = data.reminderDateTime ?? new Date();
        this.createdAt = data.createdAt ?? '';
    }
}

File: src/main/java/com/usef/workshop/todo/entity/Task.java
package com.usef.workshop.todo.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.UUID;

/**
 * @author HP
 **/
@Getter
@Setter
@Entity
@Table(name = "tasks")
@NoArgsConstructor
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE , generator = "task_seq_gen")
    @SequenceGenerator(name = "task_seq_gen", sequenceName = "task_seq", allocationSize = 1)
    private Long id;

    @Column(nullable = false, unique = true, updatable = false)
    private UUID uuid = UUID.randomUUID();

    @Column(nullable = false,unique = true)
    private String title;

    @Column(length = 1000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority;

    @ManyToOne
    @JoinColumn(nullable = false , name = "category_id")
    private Category category;

    @Column
    private Boolean completed = false;

    @Column
    private LocalDate dueDate;

    @Column
    private LocalDateTime reminderDateTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column
    private LocalDateTime updatedAt;

    @PrePersist
    protected  void onCreate(){
        this.createdAt = LocalDateTime.now();
        if (this.status == null) {
            this.status = Status.TODO;
        }
    }

    @PreUpdate
    protected void onUpdate(){
        this.updatedAt = LocalDateTime.now();
    }

    public enum Priority{
        LOW,
        MEDIUM,
        HIGH
    }

    public enum Status {
        TODO,
        IN_PROGRESS,
        DONE
    }



}
